python/lsst/daf/butler/datastore/record_data.py

```diff
@@ -46,16 +46,10 @@
 if TYPE_CHECKING:
     from ..registry import Registry
 
-# Pydantic 2 requires we be explicit about the types that are used in
```

Comment:
Would this comment still be helpful?

Reply:
Maybe; the interesting bit is _why_ `UUID` doesn't work with `Any`, and I'm guessing it's because there's nothing to distinguish it from `str`, and it doesn't see if a `str` is UUID-like unless you explicitly tell it that `UUID` is a possibility.  But that's pure speculation.

Reply:
I think this was something that became necessary after some pydantic 2 releases had already been made.

Reply:
I'll restore the comment, or something like it.